sklearn/preprocessing/label.py

```diff
@@ -472,10 +472,14 @@ def label_binarize(y, classes, neg_label=0, pos_label=1, sparse_output=False):
     classes = np.asarray(classes)
 
     if y_type == "binary":
-        if len(classes) == 1:
-            Y = np.zeros((len(y), 1), dtype=np.int)
-            Y += neg_label
-            return Y
+        if n_classes == 1:
+            if sparse_output:
+                n_classes += 1
+                classes = np.append(classes, neg_label)
```

Comment:
Yes that's better. Actually since sparse binarization is only supported for 0 `neg_label` I'll just do something like:
```python
y = column_or_1d(y)
return sp.csr_matrix((n_samples, 1), dtype=int)
```